model/semester.go

```diff
@@ -0,0 +1,33 @@
+package model
+
+type Semester struct {
+	TeachingTerm string
+	Year         int
+}
+
+// InRangeOfSemesters checks if s is between firstSemester (inclusive) and lastSemester (inclusive) or is element of semesters slice
+func (s *Semester) InRangeOfSemesters(firstSemester Semester, lastSemester Semester, semesters []Semester) bool {
+	if s == nil {
```

Comment:
nit: these emptyness (`s` can never be actualy `nil`) checks inside the function seem a bit strange in terms of API design. Are they intentional?

Reply:
Actually s can be nil so this check is necessary.

Reply:
I assume you picked this as an example (you talk about "these checks" (plural)) and you are referring to the multiple times, this is done throughout this PR.

They are/were intentional. Generally, I added these to every pointer method in this PR to ensure that anyone who uses any method added in this PR does not need to worry about whether the pointer is nil or not.

In some parts, they are even crucial to this PR working as expected, e.g. here, because I don't ever check if the user is nil (not loggedin) or not:
https://github.com/TUM-Dev/gocast/blob/cae868bde52013dd403d349c0a359804f9512380/model/user.go#L279-L281

However, in the specific code lines you commented on and a few others, it may be better to just let it panic, as this does not handle any meaningful case.

Reply:
> whether the pointer is nil or not

I hate that go does make discussing this so (unnessesarily) difficult..

The pointer can never be `nil`. The value it is pointing to can be default and thus equal to `nil`.

Having invalid input this deep into the application seems.. a bit like a smell..
But maybe that is just the rust person talking 😓

That was what I was commenting on ^^

Reply:
Suggestion is applied fc14c036963437eaefc5aaf0cef52b4313258585